### PR TITLE
PR flow for cloud

### PR DESCRIPTION
**Pull request by boekholt**
This is an initial version with additional `hardware/Makefile` targets for cloud support

### CODE REVIEW

File: hardware/setup/snap_cloud_merge.tcl [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+puts [format "%-*s %-*s %-*s %-*s"  $widthCol1 "" $widthCol2 "adding checkpoints" $widthCol3 "" $widthCol4 "[clock format [clock seconds] -format %H:%M:%S]"]
+add_files $dcp_dir/$static_region_dcp >> $log_file
+add_file $dcp_dir/$user_action_dcp >> $log_file
+set_property SCOPED_TO_CELLS {a0/action_w} [get_files $dcp_dir/$user_action_dcp] >> $log_file
+
+## 
+## linking design
+puts [format "%-*s %-*s %-*s %-*s"  $widthCol1 "" $widthCol2 "linking design" $widthCol3 "" $widthCol4 "[clock format [clock seconds] -format %H:%M:%S]"]
+link_design -mode default -reconfig_partitions {user_action} -part $fpga_part -top psl_fpga >> $log_file
+
```
Comment (kenhill) [inline hardware/setup/snap_cloud_merge.tcl]: Should `lock_design -level routing` be added for all the cells in the static region?

Comment (boekholt) [inline hardware/setup/snap_cloud_merge.tcl]: Good question - it worked for me without that. Need to do some experiments to see if we should add it.

**Comment by boekholt** [comment]
@kenhill Thanks a lot for your feedback.

Regarding the naming convention/configuration constraints: I am currently working on a new approach for SNAP configuration management (defining card type, SDRAM usage, NVMe usage, ...). When that is completed it would be a good time to look into how to handle configuration constraints for the cloud flow.

If there are no other concerns/questions besides adding `lock_design -level routing` and adding a naming convention I suggest to merge this branch into master in order to get a stake into the ground.
This would make it easier to add fixes and enhancements without having to rebase the branch again and again.

**Comment by boekholt** [comment]
Documentation needs to be added, anyway.
I'll open issues for the required "enhancements".

**Pull request merged**